### ISSUE TITLE
improve security: do not send credentials to remote server

### ISSUE THREAD

**Issue by bewest** [tidepool-org/chrome-uploader#244]
Observed: In chrome-uploader, the uploader is able to make any web requests it needs to make in order to fetch information.  In order to get carelink data, it collects Carelink user credentials in the app, where the code is running on the user's local machine.  Then, chrome-uploader sends these credentials to a remote server that is not Carelink servers.  In addition, the payload appears to be sent in the clear (although presumably over SSL transport, the message containing the passed credentials are not encrypted).

Expected: If you must collect third-party credentials, use them locally on the user's machine and delete them immediately. Eg, from chrome-uploader directly log into carelink and download the data using the permissions granted by the chrome extension.  Upload the data only, not the user's credentials.

**Comment by bewest** [comment]
It's hard to tell how logic flows through the app, it could be that some other method is being used to collect carelink data (I see also a cli tool).  As-is, this seem to be the problematic code:
* https://github.com/tidepool-org/platform-client/blob/1c24f402936948d57807aab21510c3a3faa6be16/index.js#L489
* https://github.com/tidepool-org/chrome-uploader/blob/master/lib/core/api.js#L365-L374

There's some additional confusion due to the method being named "upload" but actually retrieving data from Carelink's servers, first?

**Comment by HowardLook** [comment]
Hi Ben!

It was great to see you in person on Monday. Thanks so much for filing this issue. As discussed, we've examined this and while there are pros and cons to the current implementation, it is not a security issue. Credentials are passed to the server over TLS and as can be seen in the source are not stored or logged. Tidepool uses TLS/SSL with an up-to-date SHA2 certificate ensuring that credentials are encrypted at all times while in transit. Credentials only exist ephemerally on the server while the CSV is being fetched and are never stored to disk or retained in any way.

The current method of fetching the CSV file from CareLink is a historical artifact of our original implementation which did the fetch and parse on the server. When we created the Tidepool Uploader, we chose the most expeditious route to deliver CareLink support to our users, which is the current implementation.

It is certainly not optimal to fetch the CSV on the server and then pass it to the client for parsing, then re-upload the data. However, as stated above, it is not a security issue. For now, since we track issues in Trello, not GitHub, I've created [this](https://trello.com/c/RJ1xn52B) card and am closing this.

Feel free to call if you'd like to discuss further.

Thanks again!
Howard
